repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala

```diff
@@ -54,9 +54,9 @@ object Main extends Logging {
   // Visible for testing
   private[repl] def doMain(args: Array[String], _interp: SparkILoop): Unit = {
     interp = _interp
-    val jars = conf.getOption("spark.jars")
-      .map(_.replace(",", File.pathSeparator))
-      .getOrElse("")
+    val sparkJars = conf.getOption("spark.jars")
+    val yarnJars = conf.getOption("spark.yarn.dist.jars")
```

Comment:
Similar to the `SparkContext` check, this probably should only be done is `conf.get("spark.master")` is "yarn".